src/component/calculator.js

```diff
@@ -1,33 +1,31 @@
-import React from 'react';
+/* eslint-disable max-len */
+import React, { useState } from 'react';
 import { decodeHTMLEntities } from '../helper/util';
 import calculate from '../logic/calculate';
-// eslint-disable-next-line react/prefer-stateless-function
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      total: 0,
-      next: null,
-      operation: null,
-    };
-    this.handleClick = this.handleClick.bind(this);
-  }
 
-  handleClick = (e) => {
-    const calculateObj = { ...this.state };
+function Calculator() {
```

Comment:
- [ ] I can see that you put a lot of effort into your project, however because the purpose is to understand how hooks operate, I believe implementing the arrow function rather than the class function is a better approach. This is how I believe your syntax should be written.  :point_down:  :wink: 
```
// ./src/components/Calculator.js

import React from 'react';

const Calculator = () => (
    // reference as a js object
      <div>
       // Calculator app design goes here
      </div>
);

export default Calculator; // export Calculator component
```